src/git.ts

```diff
@@ -39,12 +39,16 @@ export class Commit {
     }
 
     /** The first line of the commit message */
-    public get subject(): string {
+    public get subject(): string | undefined {
         return this.message && this.message.split('\n', 1)[0]
     }
 
-    constructor(options?: { sha1?: string; message?: string; tasks?: TaskList }) {
-        Object.assign(this, options)
+    constructor(options: { sha1: string; message?: string; tasks?: TaskList }) {
+        this.sha1 = options.sha1
+        this.message = options.message
+        if (options.tasks) {
+            this.tasks = options.tasks
+        }
```

Comment:
would look nicer with `this.tasks = options.tasks || new TaskList()` (removing the property initialiser)